prow/_hack/create_user.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# Copyright 2023 The Jetstack contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Change USERNAME to your username (e.g. client or crierclient)
+USERNAME=crierclient
+
+CSR_FILE=$USERNAME.csr
+KEY_FILE=$USERNAME.key
+CRT_FILE=$USERNAME.crt
+
+openssl genrsa -out $KEY_FILE 2048
+openssl req -new -key $KEY_FILE -out $CSR_FILE -subj "/CN=$USERNAME"
+
+cat <<EOF | kubectl create -f -
+apiVersion: certificates.k8s.io/v1
+kind: CertificateSigningRequest
+metadata:
+    name: $USERNAME
+spec:
+    signerName: kubernetes.io/kube-apiserver-client
+    groups:
+    - system:authenticated
+    request: $(cat $CSR_FILE | base64 | tr -d '\n')
```

Comment:
suggestion (non-blocking): This kind of thing gives me the heebie-jeebies. It's so easy for a command to result in invalid YAML, or - worse - YAML which has a different meaning to what was intended. If that isn't possible today, it could easily happen in a future version where this command subtly changes. Unix pipes deal with text and YAML isn't regular text.

I'd vote to use `yq` instead to be sure that the variable is substituted properly (or use `jq` and pass in a JSON blob). Since this is a script which won't be used often, I guess it doesn't matter a great deal and this isn't a blocker!

Reply:
Thanks for the suggestion, I generally agree, however I would like to not depend on yq/ jq. This case seems somewhat safe-ish, it is a one-line base64 blob, but I agree that this can go wrong pretty quickly.